getting-started/CONCEPTS/USER-AUTH.md

```diff
@@ -0,0 +1,179 @@
+
+The `text` property is the titie and the `text` property is the body of the sign in card sent to the user.
+
+### Auto sign in
+
+With this configuration, the bot will attempt to authenticate the user when they try to interact with it. To control when for which incoming activities the bot should authenticate the user, you can specify configure the auto sign in property in the options.
+
+**C#**
+```cs
+options.AutoSignIn = (ITurnContext turnContext, CancellationToken cancellationToken) =>
+ {
+     string command = (turnContext.Activity.Value as JObject).Value<string>("commandId");
+     bool signOutActivity = command == "signOutCommand";
+     if (signOutActivity)
+     {
+         return Task.FromResult(true);
+     }
+     return Task.FromResult(false);
+ };
+```
+
+**JavaScript**
+```ts
+.withAuthentication(adapter, {
+    // settings: {
+    //     graph: {
+    //         connectionName: process.env.OAUTH_CONNECTION_NAME ?? '',
+    //         title: 'Sign in',
+    //         text: 'Please sign in to use the bot.',
+    //         endOnInvalidMessage: true
+    //     }
+    // },
+    autoSignIn: (context: TurnContext) => {
+        const signOutActivity = context.activity?.value.commandId === 'signOutCommand';
+        if (signOutActivity) {
+            return Promise.resolve(false);
+        }
+
+        return Promise.resolve(true);
+    }
+})
+```
+
+The `autoSignIn` property takes a callback that triggers the sign in flow if it returns true. It depends on the turn context from which the incomming activity details can be extracted. In the above example, the library will not attempt to sign the user in if the incomming activity `commandId` is *"signOutCommand"*.
+
+This is useful if the user should be signed in by default before attempting to interacting with the bot in general. 
+
+### Manual Sign In
+
+If the user should only be authenticated in certain scenarios then you can disable auto sign in by having the callback alway return false and trigger authentication manually.
+
+Here's an example of manually triggering sign in flow in an activity or action handler:
+
+
+**C#**
+```cs
+string? token = await app.GetTokenOrStartSignInAsync(turnContext, turnState, "graph", cancellationToken);
+if (token == null || token.Length == 0)
+{
+    await turnContext.SendActivityAsync("You have to be signed in to fulfill this request. Starting sign in flow...");
+}
+```
+
+**Javascript**
+```ts
+const token = await app.getTokenOrStartSignIn(context, state, 'graph');
+if (!token) {
+    await context.sendActivity('You have to be signed in to fulfill this request. Starting sign in flow...');
+}
+```
+
+The `app.getTokenOrStartSignIn` method will attempt to get the access token if the user is already signed in. Otherwise, the sign in flow will be triggered. The string `'graph'` references the connection name set by the user in the `settings` object of the authentication options.
```

Comment:
```suggestion
The `app.getTokenOrStartSignIn` method will attempt to get the access token if the user is already signed in. Otherwise, the sign in flow will be triggered. The string `'graph'` below references the connection name set by the user in the `settings` object of the authentication options.
```